### PR TITLE
Checkstyle: prevent using unshaded Guava classes in Arrow and AssertJ

### PR DESCRIPTION
**Pull request by jackye1995**
add new checkstyle rules for unshaded Guava classes for recently introduced libraries. @kbendick

### CODE REVIEW

File: .baseline/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -189,6 +189,16 @@
             <property name="illegalPkgs" value="com.google.common"/>
             <message key="import.illegal" value="Use org.apache.iceberg.relocated.* classes from bundled-guava module instead."/>
         </module>
+        <module name="IllegalImport">
+            <property name="id" value="BanUnrelocatedGuavaClasses"/>
```
Comment (kbendick) [inline .baseline/checkstyle/checkstyle.xml:193]: Nit: The value here looks like i was copy pasted. Should maybe be `BanUnrelocatedAssertjUtilClasses`.

File: .baseline/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -189,6 +189,16 @@
...
+            <property name="illegalClasses" value="org.assertj.core.util"/>
+            <message key="import.illegal" value="Use org.apache.iceberg.relocated.* classes from bundled-guava module instead."/>
+        </module>
+        <module name="IllegalImport">
+            <property name="id" value="BanUnrelocatedGuavaClasses"/>
```
Comment (kbendick) [inline .baseline/checkstyle/checkstyle.xml:198]: Same here, but `BanUnrelocatedArrowUtilClasses`.

File: .baseline/checkstyle/checkstyle.xml [hunk 3]
```diff
@@ -189,6 +189,16 @@
             <property name="illegalPkgs" value="com.google.common"/>
             <message key="import.illegal" value="Use org.apache.iceberg.relocated.* classes from bundled-guava module instead."/>
         </module>
+        <module name="IllegalImport">
+            <property name="id" value="BanUnrelocatedGuavaClasses"/>
+            <property name="illegalClasses" value="org.assertj.core.util"/>
+            <message key="import.illegal" value="Use org.apache.iceberg.relocated.* classes from bundled-guava module instead."/>
```
Comment (kbendick) [inline .baseline/checkstyle/checkstyle.xml:195]: Comment still references guava.

File: .baseline/checkstyle/checkstyle.xml [hunk 4]
```diff
@@ -189,6 +189,16 @@
...
+        </module>
+        <module name="IllegalImport">
+            <property name="id" value="BanUnrelocatedGuavaClasses"/>
+            <property name="illegalClasses" value="org.apache.arrow.util"/>
+            <message key="import.illegal" value="Use org.apache.iceberg.relocated.* classes from bundled-guava module instead."/>
```
Comment (kbendick) [inline .baseline/checkstyle/checkstyle.xml:200]: Same note on comment.

**Review by kbendick — APPROVED** [review]
A few updates needed in the checkstyle descriptions and comments, but this is a good catch. 

Is there anything in `org.apache.arrow.util` we might ever need? 

I think it's safe to add this and then it can be reconsidered or made more explicit to the Preconditions check otherwise.

I know that my Intellij autoimports `Preconditions` from all sorts of places and it's easy to miss.

**Comment by kbendick** [comment]
cc @nastra who often updates the check style file, but I think this is safe and will help accidentally importing Preconditions and other Guava / google commons things from these libraries that also seem to have forked.

File: .baseline/checkstyle/checkstyle.xml [hunk 5]
```diff
@@ -189,6 +189,16 @@
             <property name="illegalPkgs" value="com.google.common"/>
             <message key="import.illegal" value="Use org.apache.iceberg.relocated.* classes from bundled-guava module instead."/>
         </module>
+        <module name="IllegalImport">
+            <property name="id" value="BanUnrelocatedGuavaClasses"/>
+            <property name="illegalClasses" value="org.assertj.core.util"/>
```
Comment (kbendick) [inline .baseline/checkstyle/checkstyle.xml:194]: Should this be `illegalPkgs` like on line 189?

File: .baseline/checkstyle/checkstyle.xml [hunk 6]
```diff
@@ -189,6 +189,16 @@
...
+            <message key="import.illegal" value="Use org.apache.iceberg.relocated.* classes from bundled-guava module instead."/>
+        </module>
+        <module name="IllegalImport">
+            <property name="id" value="BanUnrelocatedGuavaClasses"/>
+            <property name="illegalClasses" value="org.apache.arrow.util"/>
```
Comment (kbendick) [inline .baseline/checkstyle/checkstyle.xml:199]: This one should also probably be `illegalPkgs`.

**Review by kbendick — APPROVED** [review]
I checked this out locally and run `./gradlew check` and everything looked good. 👍

File: .baseline/checkstyle/checkstyle.xml (hunk 1 above)
Comment (jackye1995) [reply .baseline/checkstyle/checkstyle.xml:193]: I was mostly following this:

```
<module name="IllegalImport">
            <property name="id" value="BanUnrelocatedGuavaClasses"/>
            <property name="illegalPkgs" value="com.google.common"/>
            <message key="import.illegal" value="Use org.apache.iceberg.relocated.* classes from bundled-guava module instead."/>
        </module>
```

I am actually not sure if the ID has to be unique, let me check

Comment (jackye1995) [reply .baseline/checkstyle/checkstyle.xml:193]: Ah I see, so that's a user-defined name, then let me update it

Comment (kbendick) [reply .baseline/checkstyle/checkstyle.xml]: I checked myself when I was verifying this (not 100% used to updating these files), and they do not have to be unique.

It's just used in the output it seems, like this example where I added an import.

```
> Task :iceberg-arrow:checkstyleMain FAILED
[ant:checkstyle] [ERROR] /Users/kylebendickson/repos/iceberg-kyle/arrow/src/main/java/org/apache/iceberg/arrow/ArrowAllocation.java:23:1: Use org.apache.iceberg.relocated.* classes from bundled-guava module instead. [BanUnrelocatedGuavaClasses]
```

**Review by nastra — COMMENTED** [review]
changes make sense, what about preventing usage of `org.apache.parquet.Preconditions`? I believe this is also being used unintentionally at a few places

**Pull request merged**